contracts/src/SafeInternationalHarbour.sol

```diff
@@ -0,0 +1,406 @@
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        bytes data
+    );
+
+    // ------------------------------------------------------------------
+    // External & public functions
+    // ------------------------------------------------------------------
+
+    /**
+     * @notice Publish a Safe transaction and/or append a signature to it.
+     *
+     * @dev If `safeTxHash` has been seen before, its parameters are *not* validated nor overwritten –
+     *      the call simply appends the `(r,s)` pair for `signer`.
+     *
+     * @param safeAddress    Target Safe Smart‑Account.
+     * @param chainId        Chain id the transaction is meant for.
+     * @param nonce          Safe nonce.
+     * @param to             Destination of the inner call/delegatecall.
+     * @param value          ETH value forwarded by the Safe.
+     * @param data           Calldata executed by the Safe.
+     * @param operation      0 = CALL, 1 = DELEGATECALL.
+     * @param safeTxGas      Gas forwarded to the inner call.
+     * @param baseGas        Fixed overhead reimbursed to the submitting signer.
+     * @param gasPrice       Gas price used for reimbursement.
+     * @param gasToken       ERC‑20 token address for refunds (`address(0)` = ETH).
+     * @param refundReceiver Address receiving the gas refund.
+     * @param signature      **Single** 65‑byte ECDSA signature.
+     *
+     * @return listIndex     Index of the stored signature in the signer‑specific list.
+     *
+     * @custom:events Emits {SignatureStored}.
+     */
+    function enqueueTransaction(
+        address safeAddress,
+        uint256 chainId,
+        uint256 nonce,
+        address to,
+        uint256 value,
+        bytes calldata data,
+        uint8 operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        bytes calldata signature
+    ) external returns (uint256 listIndex) {
+        if (signature.length != 65) revert InvalidECDSASignatureLength();
+
+        // ------------------------------------------------------------------
+        // Build the EIP‑712 digest that uniquely identifies the SafeTx
+        // ------------------------------------------------------------------
+        bytes32 safeTxHash = _computeSafeTxHash(
+            safeAddress,
+            chainId,
+            nonce,
+            to,
+            value,
+            data,
+            operation,
+            safeTxGas,
+            baseGas,
+            gasPrice,
+            gasToken,
+            refundReceiver
+        );
+
+        // Recover signer and split signature into (r,s)
+        (address signer, bytes32 r, bytes32 s) = _recoverSignerAndRS(safeTxHash, signature);
+
+        // Store parameters only once (idempotent write)
+        SafeTransaction storage slot = _txDetails[safeTxHash];
+        if (slot.to == address(0)) {
```

Comment:
This means signing for `to = address(0)` for burning ETH isn't well supported.

Why not include a `uint8 stored` flag that gets packed into the same slot as the `to` address (so it isn't additional storage) and can always be set to `1` on initial storage.

Reply:
To be honest, I thought about this, but I wanted to keep the code as simple as possible - my rationale for leaving it like this is:
1) there are plenty of other burn addresses to choose from
2) the current safe transaction service doesn't allow a zero address either

Reply:
> I wanted to keep the code as simple as possible

I don't think this is complex TBH:

```solidity
struct SafeTransaction {
    bool stored;
    address to;
    // ...
}

if (!slot.stored) {
    slot.stored = true;
    slot.to = to;
    // ...
}
```

In fact, I would argue it more clearly shows intent and uses a single storage slot anyway.

Reply:
You're right. Now I'm looking at it, and it definitely looks cleaner. Thanks!